ssr/choo.js

```diff
@@ -62,18 +62,23 @@ module.exports.listRoutes = function (app) {
 
 module.exports.render = function (app, route, cb) {
   var state = {}
+  var body = ''
   state._experimental_prefetch = []
-  app.toString(route, state)
-
-  // TODO: replace with p-wait-all, once it knows how to handle an empty array.
-  Promise.all(state._experimental_prefetch)
-    .then(render, render)
+  try {
+    body = app.toString(route, state)
+    Promise.resolve().then(function () {
+      return Promise.all(state._experimental_prefetch)
+    }).then(render, render)
+  } catch (err) {
+    // logging this to make sure people can debug the whole stack trace (sorry ui)
+    console.error(err)
```

Comment:
I'm thinking the best way would be to render the full stack trace to the browser, and otherwise keep the one-line output in the bankai ui, but couldn't quite find my way in the code to properly intercept the error.